### ISSUE TITLE
Add Support for Test Connection Feature to Validate Vector Store Connectivity

### ISSUE THREAD

**Issue by SurajKamal24** [MuleSoft-AI-Chain-Project/mule-vectors-connector#55]
Since most of the vector stores/databases are cloud-based or hosted on remote servers, it would be handy to have the test connection feature to instantly verify connectivity to vector stores/databases. This will improve debugging efficiency to confirm connectivity instantly during connector configuration in Anypoint Studio.

Test Connection in Standard Mule Connector

![test-connection](https://github.com/user-attachments/assets/8e52e28d-826b-410a-a7a8-9e2624cf4033)

**Comment by tbolis-at-mulesoft** [comment]
We are working on enhancing the connector configuration. But since langchain4j doesn't expose a ready to use method, this may require a little bit of effort on our side and increase the effort to maintain the codebase.

Anyway thank you we'll further investigate this point.

![image](https://github.com/user-attachments/assets/2dc58ebc-56dd-478a-95ae-752110c281dd)

FYI @amirkhan-ak

**Comment by tbolis-at-mulesoft** [comment]
Here I see an issue due to the fact that test connection should be implemented also to test model service. 

I tried to make Subtypes work with ConnectionProviders, but I'm facing a blocking issue... 

The test connection works fine only the first time you configure it, while if you close the modal and reopen it, the test always evaluates parameters  loaded when opening the modal. To have the test connection use the changes I had to close and reopen the modal.
